Expand use of golangci-lint linters

Aside from the "Enabled by Default" linters, we have also explicitly enabled several others in our wrapper script (perhaps to annoying effect as noted by #176).

Here are the current entries:

https://github.com/atc0005/elbow/blob/5f074f18c7527520adc7c96ffb060c142220d1c7/testing/run_linting_checks.sh#L88-L94

While reviewing the [supported linters](https://github.com/golangci/golangci-lint#supported-linters), these caught my eye:

- prealloc
- misspell
- maligned
- gocyclo
- gocognit
- dupl
- unconvert
- golint
- gocritic

When ran against current master (5f074f18c7527520adc7c96ffb060c142220d1c7), these issues are noted:

<details>
  <summary>Verbose linting output</summary>
<pre>
Calling wrapper script: bash testing/run_linting_checks.sh
config/merge_incomplete_configs_test.go:225:9: `Explictly` is a misspelling of `Explicitly` (misspell)
        t.Log("Explictly setting AppMetadata fields to pass validation checks")
               ^
config/merge_incomplete_configs_test.go:352:9: `Explictly` is a misspelling of `Explicitly` (misspell)
        t.Log("Explictly setting AppMetadata fields to pass validation checks")
               ^
config/validate_test.go:29:1: cognitive complexity 177 of func `TestValidate` is high (> 30) (gocognit)
func TestValidate(t *testing.T) {
^
config/compare.go:49:1: cognitive complexity 40 of func `CompareConfig` is high (> 30) (gocognit)
func CompareConfig(got Config, wanted Config, t *testing.T) {
^
paths/paths.go:168:1: cognitive complexity 32 of func `ProcessPath` is high (> 30) (gocognit)
func ProcessPath(config *config.Config, path string) (matches.FileMatches, error) {
^
config/get.go:164:10: string `text` has 4 occurrences, make it a constant (goconst)
                return "text"
                       ^
config/config_test.go:54:14: string `.exe` has 3 occurrences, make it a constant (goconst)
                appName += ".exe"
                           ^
config/config.go:413:15: string `NotSet` has 4 occurrences, make it a constant (goconst)
                c.AppName = "NotSet"
                            ^
config/config_test.go:52:13: string `elbow` has 3 occurrences, make it a constant (goconst)
        appName := "elbow"
                   ^
config/config_test.go:53:21: string `windows` has 3 occurrences, make it a constant (goconst)
        if runtime.GOOS == "windows" {
                           ^
config/validate.go:24:1: cyclomatic complexity 40 of func `(Config).Validate` is high (> 30) (gocyclo)
func (c Config) Validate() error {
^
main.go:141:6: sloppyLen: len(fileMatches) <= 0 can be len(fileMatches) == 0 (gocritic)
                if len(fileMatches) <= 0 {
                   ^
Non-zero exit code from golangci-lint: 1
Linting failed, most recent failure: golangci-lint
Makefile:88: recipe for target 'linting' failed
make: *** [linting] Error 1
</pre>
</details>

Some are already noted (#176), others might be too specific for me to deal with right now (`gocyclo` and `gocognit`), but these noted issues appear to be easily reviewed/fixed and worth enabling now:

```
main.go:141:6: sloppyLen: len(fileMatches) <= 0 can be len(fileMatches) == 0 (gocritic)
                if len(fileMatches) <= 0 {
                   ^
config/merge_incomplete_configs_test.go:225:9: `Explictly` is a misspelling of `Explicitly` (misspell)
        t.Log("Explictly setting AppMetadata fields to pass validation checks")
               ^
config/merge_incomplete_configs_test.go:352:9: `Explictly` is a misspelling of `Explicitly` (misspell)
        t.Log("Explictly setting AppMetadata fields to pass validation checks")
               ^
```

--- a/testing/run_linting_checks.sh
+++ b/testing/run_linting_checks.sh
@@ -91,7 +91,13 @@ else
         -E stylecheck \
         -E goconst \
         -E depguard \
-        -E prealloc
+        -E prealloc \
+        -E misspell \
+        -E maligned \
+        -E dupl \
+        -E unconvert \
+        -E golint \
+        -E gocritic
 fi
 
 status=$?

